cpu/gd32v/periph/adc.c

```diff
@@ -122,7 +124,7 @@ int adc_init(adc_t line)
     }
     /* set clock prescaler to get the maximal possible ADC clock value */
     for (clk_div = 2; clk_div < 8; clk_div += 2) {
-        if ((CLOCK_CORECLOCK / clk_div) <= MAX_ADC_SPEED) {
+        if ((CLOCK_CORECLOCK / clk_div) <= ADC_CLK_MAX) {
```

Comment:
```suggestion
        if ((periph_apb_clk(APB2) / clk_div) <= ADC_CLK_MAX) {
```

Shouldn't this also be applied here, btw.? The code indicates that the ADC for the GD32V is also clocked by ABP2

Reply:
I am not familiar with, nor have access to, GigaDevices parts. I'll accept the change if you are familiar enough to say this is how they are clocked.

Reply:
The GD32V is more or less a knock-off of STM32F1 but with the ARM CPU replaced by RISC-V. The peripherals are match the STM32F1 almost perfectly.

Maybe @bergzand or @gschorcht can confirm this?

Otherwise I would be fine to only fix STM32 for now.

Reply:
Yes, the peripherals of the GD32VF103 are almost identical to those of the STM32F1. The ADC is clocked by APB2, which actually corresponds to `CLOCK_CORECLOCK`. Therefore, using `CLOCK_CORECLOCK` is not wrong here, but using `periph_apb_clk(APB2)` is the cleaner solution.

Reply:
Suggested change made.